drivers/wifi/eswifi/eswifi_socket.c

```diff
@@ -132,9 +132,18 @@ static void eswifi_off_read_work(struct k_work *work)
 	}
 
 	__select_socket(eswifi, socket->index);
+	
+	/* Verify if we can allocate a rx packet before reading data to prevent leaks */
+	pkt = net_pkt_rx_alloc_with_buffer(eswifi->iface, 1460,
```

Comment:
So this means that we always try to allocate the "full" buffer regardless of how much data there is, this could be wasting lot of net_buf's.
Is it possible to figure out how much data there is to read (without actually reading it), and then allocate just that amount?

Reply:
Absolutely, I’ll check tomorrow if there is an API to check the size of the pending packet or if we can resize the buffer we have allocated. Thanks !

Reply:
Hi, I pushed a new commit to fix the issue of allocation. My strategy is to allocate the buffer with MTU, then read and then resize with the API net_pkt_update_length() so that we allocate the returned length of read function.

Reply:
Ok, I think I did a mistake, net_pkt_update_length() only modify length attribute of net_pkt, I'll try to resize the pkt  by creating a new pkt to the exact size thanks to read return and then copy the old pkt into the new pkt and then unref the old pkt